File: CSE220/Python Based Projects/Fall 22 MidTerm Solve/Question1.py
def nonPrimeCircular(arr, size):
	nonPrime = [0]*len(arr)
	nonPrimeCount = 0;
	for i in range(size):
		divNum = 0;
		temp = arr[i];
		while temp != 0:
			if arr[i] % temp == 0:
				divNum += 1;
			temp -= 1;
		if divNum > 2:
			nonPrime[nonPrimeCount] = arr[i];
			nonPrimeCount += 1
	
	if nonPrimeCount == 0:
		print("All are prime")
	else:
		cir = [0]*nonPrimeCount
		idx = len(cir) - 1;
		for i in range(nonPrimeCount-1, -1, -1):
			cir[idx] = nonPrime[i]
			idx = (idx + 1) % len(cir)
		return cir


lin = [7, 1, 5, 15, 52, 26, 2, 3, 0, 0]
cir = nonPrimeCircular(lin, 8)
print()
print(cir, end="\n\n")

lin = [7, 1, 5, 2, 3, 0, 0]
cir = nonPrimeCircular(lin, 5)
print(cir)

File: CSE220/Fall 22 Mid Solve/Question1.java
public class Question1 {

	public static void main(String[] args) {
		Question1 q1 = new Question1();
		
		int[] arr1 = { 7, 1, 5, 15, 52, 26, 2, 3, 0, 0 };
		int[] npcArr = q1.nonPrimeCircular(arr1, 8);
		q1.fullLinearPrint(npcArr);
		
		int[] arr2 = { 7, 1, 5, 2, 3, 0, 0 };
		npcArr = q1.nonPrimeCircular(arr2, 5);
		q1.fullLinearPrint(npcArr);

	}

	public void fullLinearPrint(int[] arr) {
		if (arr.length == 0) {
			System.out.println("[]");
		} else if (arr.length == 1) {
			System.out.print("[" + arr[0] + "]");
		} else {
			System.out.print("[" + arr[0] + ", ");
			for (int i = 1; i < arr.length - 1; i++) {
				System.out.print(arr[i] + ", ");
			}
			System.out.print(arr[arr.length - 1] + "]\n");
		}
	}

	public int[] nonPrimeCircular(int[] arr, int size) {
		int[] nonPrime = new int[arr.length];
		int nonPrimeCount = 0;
		for (int i = 2; i < size - 1; i++) {
			boolean divFound = false;
			int temp = arr[i] - 1;
			while (temp > 1) {
				if (arr[i] % temp == 0) {
					divFound = true;
					break;
				}
				temp--;
			}
			if (divFound == true) {
				nonPrime[nonPrimeCount] = arr[i];
				nonPrimeCount++;
			}
		}
		if (nonPrimeCount == 0) {
			System.out.println("All are prime");
			return new int[0];
		} else {
			int[] cir = new int[nonPrimeCount];
			int idx = cir.length - 1;
			for (int i = nonPrimeCount - 1; i >= 0; i--) {
				cir[idx] = nonPrime[i];
				idx = (idx + 1) % cir.length;
			}
			return cir;
		}
	}
}